src/mappings/nabla/routerEventHandler.ts

```diff
@@ -0,0 +1,150 @@
+import { hexToSs58, ss58ToHex } from './addresses'
+
+export async function handleRouterEvent(
+    ctx: EventHandlerContext,
+    router: Router
+) {
+    const event = decodeEvent(ctx.event.args.data)
+
+    switch (event.__kind) {
+        case 'OwnershipTransferred':
+            // This event will always be emitted on router creation
+            // no action required
+            break
+
+        case 'Paused':
+            await handlePaused(ctx, router)
+            break
+
+        case 'Swap':
+            await handleSwap(ctx, event, router)
+            break
+
+        case 'SwapPoolRegistered':
+            await handleSwapPoolRegistered(ctx, event, router)
+            break
+
+        case 'SwapPoolUnregistered':
+            await handleSwapPoolUnregistered(ctx, event, router)
+            break
+
+        case 'Unpaused':
+            await handleUnpaused(ctx, router)
+            break
+    }
+}
+
+export async function handlePaused(ctx: EventHandlerContext, router: Router) {
+    // Set the 'paused' property to true
+    router.paused = true
+    await ctx.store.save(router)
+}
+
+export async function handleUnpaused(ctx: EventHandlerContext, router: Router) {
+    // Set the 'paused' property to false
+    router.paused = false
+    await ctx.store.save(router)
+}
+
+export async function handleSwap(
+    ctx: EventHandlerContext,
+    event: Event_Swap,
+    router: Router
+) {
+    const tokenIn = await getOrCreateNablaToken(ctx, event.tokenIn)
+    const tokenOut = await getOrCreateNablaToken(ctx, event.tokenOut)
+
+    const swapPoolIn = await getSwapPoolsOfRouterForToken(
+        ctx,
+        router.id,
+        tokenIn.id
+    )
+    const swapPoolOut = await getSwapPoolsOfRouterForToken(
+        ctx,
+        router.id,
+        tokenOut.id
+    )
+
+    if (swapPoolIn !== undefined) {
+        await updateSwapPoolCoverageAndSupply(ctx, swapPoolIn)
+        await ctx.store.save(swapPoolIn)
+    }
+
+    if (swapPoolOut !== undefined) {
+        await updateSwapPoolCoverageAndSupply(ctx, swapPoolOut)
+        await ctx.store.save(swapPoolOut)
+    }
+}
+
+export async function handleSwapPoolRegistered(
+    ctx: EventHandlerContext,
+    event: Event_SwapPoolRegistered,
+    router: Router
+) {
+    const swapPoolSs58Address = hexToSs58(event.pool)
+    const swapPool = await getSwapPool(ctx, swapPoolSs58Address)
+
+    if (swapPool === undefined) {
+        // this is a non-standard or malicious swap pool, ignore
+        return
+    }
+
+    const token = await getOrCreateNablaToken(ctx, ss58ToHex(swapPool.token.id))
+
+    const registeredSwapPool = await getSwapPoolsOfRouterForToken(
+        ctx,
+        router.id,
+        swapPool.token.id
+    )
+
+    if (registeredSwapPool) {
+        if (registeredSwapPool.id === swapPool.id) {
+            // this swap pool is already registered, no action required
+            return
+        }
+
+        registeredSwapPool.router = null
```

Comment:
I think I followed the original logic of the Ethereum based indexer of the Nabla team. It might be better to keep it as this is a scenario that would happen whenever we need to adapt the slippage curve (e.g., we see that slippage is too low or too high in practive). The slippage curve of a swap pool is immutable by design and one would deploy and attach a new swap pool for the same token but a different slippage curve. After that liquidity providers would need to withdraw liquidity from the old pool and deposit into the new pool.

We don't have a UI for this yet (makes sense to only implement once we need it) but for that case we would need to keep the old swap pool in the indexer.